mdoc/src/main/scala-3/mdoc/internal/markdown/CodePrinter.scala

```diff
@@ -23,8 +23,10 @@ class CodePrinter(ps: PrintStream, baseIndent: Int = 0, baseNest: Int = 0) {
     this
   }
 
-  def appendLines(body: String) = {
+  def appendLines(body: String, nest: Boolean = false) = {
+    if (nest) nestCount += 1
     body.linesIterator.toArray.foreach(this.println)
+    if(nest) nestCount -= 1
```

Comment:
For my own mental note - will need to keep that in mind when doing #498, to make sure that generated definition is wrapped in `{}` on scala 2, as indentation alone won't cut it.